#! /bin/sh /usr/share/dpatch/dpatch-run
## 032_suexec_is_shared by Adam Conrad <adconrad@0c3.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch to allow suEXEC as shared.

@DPATCH@
--- apache2/os/unix/unixd.c	2004-09-27 19:53:29.000000000 -0600
+++ apache2/os/unix/unixd.c	2004-09-27 19:56:57.000000000 -0600
@@ -209,6 +209,10 @@
 
     /* Check for suexec */
     unixd_config.suexec_enabled = 0;
+    /* If mod_suexec isn't linked in, we shouldn't test for the binary */
+    if (ap_find_linked_module("mod_suexec.c") == NULL) {
+	return;
+    }
     if ((apr_stat(&wrapper, SUEXEC_BIN,
                   APR_FINFO_NORM, ptemp)) != APR_SUCCESS) {
         return;
